File: app/src/main/java/tpcreative/co/qrscanner/ui/create/GenerateFragment.java
package tpcreative.co.qrscanner.ui.create;
import android.content.Context;
import android.os.Bundle;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import androidx.annotation.Nullable;
import com.jaychang.srv.SimpleRecyclerView;
import java.util.ArrayList;
import java.util.List;
import butterknife.BindView;
import tpcreative.co.qrscanner.R;
import tpcreative.co.qrscanner.common.BaseFragment;
import tpcreative.co.qrscanner.common.Navigator;
import tpcreative.co.qrscanner.common.Utils;
import tpcreative.co.qrscanner.common.services.QRScannerApplication;
import tpcreative.co.qrscanner.model.QRCodeType;

public class GenerateFragment extends BaseFragment implements GenerateCell.ItemSelectedListener,GenerateView {

    private static final String TAG = GenerateFragment.class.getSimpleName();
    @BindView(R.id.recyclerView)
    SimpleRecyclerView recyclerView;
    private GeneratePresenter presenter;

    public static GenerateFragment newInstance(int index) {
        GenerateFragment fragment = new GenerateFragment();
        Bundle b = new Bundle();
        b.putInt("index", index);
        fragment.setArguments(b);
        return fragment;
    }

    @Override
    protected int getLayoutId() {
        return 0;
    }

    @Override
    protected View getLayoutId(LayoutInflater inflater, ViewGroup viewGroup) {
        View view = inflater.inflate(R.layout.fragment_generate, viewGroup, false);
        return view;
    }

    @Override
    protected void work() {
        super.work();
        presenter = new GeneratePresenter();
        presenter.bindView(this);
        presenter.setList();
    }

    @Override
    public void setMenuVisibility(boolean menuVisible) {
        super.setMenuVisibility(menuVisible);
        if (menuVisible) {
            QRScannerApplication.getInstance().getActivity().onShowFloatingButton(GenerateFragment.this);
            Utils.Log(TAG, "isVisible");
        } else {
            Utils.Log(TAG, "isInVisible");
        }
    }

    @Nullable
    @Override
    public Context getContext() {
        return super.getContext();
    }

    @Override
    public void onSetView() {
        bindData();
    }

    private void bindData() {
        List<QRCodeType> Galaxys = presenter.mList;
        //CUSTOM SORT ACCORDING TO CATEGORIES
        List<GenerateCell> cells = new ArrayList<>();
        //LOOP THROUGH GALAXIES INSTANTIATING THEIR CELLS AND ADDING TO CELLS COLLECTION
        for (QRCodeType galaxy : Galaxys) {
            GenerateCell cell = new GenerateCell(galaxy);
            cell.setListener(this);
            cells.add(cell);
        }
        recyclerView.addCells(cells);
    }

    @Override
    public void onClickItem(int position, boolean isChecked) {
        if (!Utils.isPremium()){
            position = position + 1;
        }
        switch (position){
            case 0:{
                Navigator.onGenerateView(getActivity(),null,BarcodeFragment.class);
                break;
            }
            case 1:{
                Navigator.onGenerateView(getActivity(),null,EmailFragment.class);
                break;
            }
            case 2 :{
                Navigator.onGenerateView(getActivity(),null,MessageFragment.class);
                break;
            }
            case 3 :{
                Navigator.onGenerateView(getActivity(),null,LocationFragment.class);
                break;
            }case 4:{
                Navigator.onGenerateView(getActivity(),null,EventFragment.class);
                break;
            }
            case 5:{
                Navigator.onGenerateView(getActivity(),null,ContactFragment.class);
                break;
            }
            case 6:{
                Navigator.onGenerateView(getActivity(),null,TelephoneFragment.class);
                break;
            }
            case 7:{
                Navigator.onGenerateView(getActivity(),null,TextFragment.class);
                break;
            }
            case 8:{
                Navigator.onGenerateView(getActivity(),null,WifiFragment.class);
                break;
            }
            case 9:{
                Navigator.onGenerateView(getActivity(),null,UrlFragment.class);
                break;
            }
        }
    }

    @Override
    public void onClickShare(String value) {

    }

    @Override
    public void onStop() {
        super.onStop();
        Utils.Log(TAG,"onStop");
    }

    @Override
    public void onStart() {
        super.onStart();
        Utils.Log(TAG,"onStart");
    }

    @Override
    public void onDestroy() {
        super.onDestroy();
        Utils.Log(TAG,"onDestroy");
    }

    @Override
    public void onResume() {
        super.onResume();
        Utils.Log(TAG,"onResume");
    }

    @Override
    public void onInitView() {

    }
}

File: app/src/main/java/tpcreative/co/qrscanner/ui/create/GenerateFragment.kt
package tpcreative.co.qrscanner.ui.create
import android.content.*
import android.os.Bundle
import android.view.*
import tpcreative.co.qrscanner.R
import tpcreative.co.qrscanner.common.*
import tpcreative.co.qrscanner.databinding.FragmentGenerateBinding
import tpcreative.co.qrscanner.model.QRCodeType
import tpcreative.co.qrscanner.viewmodel.GenerateViewModel
import java.util.*

class GenerateFragment : BaseFragment(), GenerateCell.ItemSelectedListener {
    lateinit var viewModel : GenerateViewModel
    lateinit var binding : FragmentGenerateBinding
    override fun getLayoutId(): Int {
        return 0
    }

    override fun getLayoutId(inflater: LayoutInflater?, viewGroup: ViewGroup?): View {
        binding = FragmentGenerateBinding.inflate(layoutInflater)
        return binding.root
    }

    override fun work() {
        super.work()
        initUI()
    }

    override fun setMenuVisibility(menuVisible: Boolean) {
        super.setMenuVisibility(menuVisible)
    }

    override fun getContext(): Context? {
        return super.getContext()
    }

    fun bindData(data : MutableList<QRCodeType>) {
        //CUSTOM SORT ACCORDING TO CATEGORIES
        val cells: MutableList<GenerateCell?> = ArrayList()
        //LOOP THROUGH GALAXIES INSTANTIATING THEIR CELLS AND ADDING TO CELLS COLLECTION
        for (galaxy in data) {
            val cell = GenerateCell(galaxy)
            cell.setListener(this)
            cells.add(cell)
        }
        binding.recyclerView.removeAllCells()
        binding.recyclerView.addCells(cells)
    }

    override fun onClickItem(position: Int, isChecked: Boolean) {
        val mPosition = position
        when (mPosition) {
            0 -> {
                Navigator.onGenerateView(activity, null, BarcodeActivity::class.java)
            }
            1 -> {
                Navigator.onGenerateView(activity, null, EmailActivity::class.java)
            }
            2 -> {
                Navigator.onGenerateView(activity, null, MessageActivity::class.java)
            }
            3 -> {
                Navigator.onGenerateView(activity, null, LocationActivity::class.java)
            }
            4 -> {
                Navigator.onGenerateView(activity, null, EventActivity::class.java)
            }
            5 -> {
                Navigator.onGenerateView(activity, null, ContactActivity::class.java)
            }
            6 -> {
                Navigator.onGenerateView(activity, null, TelephoneActivity::class.java)
            }
            7 -> {
                Navigator.onGenerateView(activity, null, TextActivity::class.java)
            }
            8 -> {
                Navigator.onGenerateView(activity, null, WifiActivity::class.java)
            }
            9 -> {
                Navigator.onGenerateView(activity, null, UrlActivity::class.java)
            }
        }
    }

    override fun onClickShare(value: String?) {}
    override fun onStop() {
        super.onStop()
        Utils.Log(TAG, "onStop")
    }

    override fun onStart() {
        super.onStart()
        Utils.Log(TAG, "onStart")
    }

    override fun onDestroy() {
        super.onDestroy()
        Utils.Log(TAG, "onDestroy")
    }

    override fun onResume() {
        super.onResume()
        Utils.Log(TAG, "onResume")
    }

    companion object {
        private val TAG = GenerateFragment::class.java.simpleName
        fun newInstance(index: Int): GenerateFragment {
            val fragment = GenerateFragment()
            val b = Bundle()
            b.putInt("index", index)
            fragment.arguments = b
            return fragment
        }
    }
}